File: Java/DizzyOnABridge.java
public class DizzyOnABridge {
    public static void main(String[] args) {
        int bridgeSize = 15;
        int manPosition = (int) ((bridgeSize + 1) / 2);
        int dracula = 0;
        while (true) {
            double randomNumber = Math.random();
            if (randomNumber < 0.5) manPosition--;
            else manPosition++;

            for (int dvacula = 1; dvacula <= bridgeSize; dvacula++) {
                if (dvacula == manPosition) System.out.print("o");
                else System.out.print("-");
            }
            System.out.print("\n");

            if (manPosition < 1 || manPosition > bridgeSize) break;
            dracula++;
        }
        System.out.println("The man moved forward " + dracula + " times " +
                           "before falling off the bridge.");
    }
}

File: Python/DizzyOnABridge.py
import random

bridgeLength = 5
manPosition = 3
moveCount = 0

print ("|--o--|")
while True:
    rand = random.random()
    if (rand < 0.5):
        manPosition -= 1
    else:
        manPosition += 1

    toPrint = "|"
    for i in range(1, bridgeLength + 1):
        if (i == manPosition):
            toPrint += "o"
        else:
            toPrint += "-"

    toPrint += "|"

    print(toPrint)
    moveCount += 1

    if (manPosition == 0 or manPosition == 6):
        print("The man moved " + str(moveCount) + " times before falling off.")
        break
